internal/dag/builder_test.go

```diff
@@ -6945,7 +6945,14 @@ func TestDAGInsert(t *testing.T) {
 			objs: []interface{}{
 				proxyMultipleBackends, s2,
 			},
-			want: listeners(),
+			want: listeners(
```

Comment:
Why does this now return listeners?

Reply:
I think this is according to [design doc chapter:](https://github.com/projectcontour/contour/blob/main/design/httpproxy-conflict.md#multiple-services) 

> For the case where multiple services are referenced from a Spec.Route of an HTTPProxy and has an Error, Contour will set the invalid service weight to be zero so that no traffic is routed to the service in error and will also set an error in the Conditions of the object.

That is, since one (of the two) service references is still valid, traffic should be routed to the valid service.